src/tests/TouchFree.test.ts

```diff
@@ -78,9 +79,68 @@ describe('TouchFree', () => {
         expect(TouchFree.GetCurrentCursor()).toBe(newCursor);
     });
 
-    it('GetInputController should get the input controller correctly', () => {
+    test('GetInputController should get the input controller correctly', () => {
         TouchFree.Init();
         const controller = TouchFree.GetInputController();
         expect(controller instanceof WebInputController).toBe(true);
     });
+
+    test('ControlAnalyticsSession should call AnalyticsSessionRequest with the correct arguments', () => {
+        ConnectionManager.init();
+        const serviceConnection = ConnectionManager.serviceConnection();
+        if (!serviceConnection) fail('Service connection not available');
+
+        const applicationName = 'testApplication';
+
+        const testFn = jest
+            .spyOn(serviceConnection, 'AnalyticsSessionRequest')
+            .mockImplementation((requestType, sessionID, callback) => {
+                expect(sessionID.includes(applicationName)).toBe(true);
+                callback?.(new WebSocketResponse('test', 'Success', 'test', 'test'));
```

Comment:
👀 This callback isn't tested. Should we also test with passing a callback?

Reply:
That callback is just to mock the service responding to trigger the ID set. Will put a comment there for it

Reply:
Ah true it should be tested in `AnalyticsSessionRequest` rather than here